### PR TITLE
Fixes parenthesis when \n is involved

### PR DESCRIPTION
**Pull request by Warxcell**
```
JSON_CONTAINS(p.state, :positive_value, :ordered_state) = :positive_value 
OR 
JSON_CONTAINS(p.state, :positive_value, :completed_state) = :positive_value)
```

when we have OR like that (without space before OR) - it doesn't add parenthesis.

### CODE REVIEW

**Comment by greg0ire** [comment]
I guess there is the same issue with tabulations, right?

**Comment by Warxcell** [comment]
> I guess there is the same issue with tabulations, right?

I don't know. Who uses tabs on queries anyways?

File: lib/Doctrine/ORM/Query/Expr/Composite.php [hunk 1]
```diff
@@ -63,7 +66,8 @@ private function processQueryPart($part)
         }
 
         // Fixes DDC-1237: User may have added a where item containing nested expression (with "OR" or "AND")
-        if (stripos($queryPart, ' OR ') !== false || stripos($queryPart, ' AND ') !== false) {
+        $singleLineQueryPart = str_replace(PHP_EOL, ' ', $queryPart);
+        if (stripos($singleLineQueryPart, ' OR ') !== false || stripos($singleLineQueryPart, ' AND ') !== false) {
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Query/Expr/Composite.php:70]: Maybe we would be better off with one single `preg_replace` here, with `\s(OR|AND)\s`. That way tabs are handled, and line feeds are preserved, that's what I'm getting at.

Comment (Warxcell) [reply lib/Doctrine/ORM/Query/Expr/Composite.php:70]: Yeah, but maybe `str_replace` is faster. I will try `preg_replace`.

**Review by greg0ire — APPROVED** [review]
> Yeah, but maybe str_replace is faster. If we will use RegEx, I can just use preg_match and not replace anything.

Probably, but here I don't think we are in the hot path: we are in dql parsing which:
- is cached
- does not happen a lot (a few queries vs other code that applies on every row of the result set)

**Comment by Warxcell** [comment]
> * is cached
> * does not happen a lot (a few queries vs other code that applies on every row of the result set)

I think it's not the actual flow.

This is QueryBuilder, which builds DQL which is then parsed and after that cached.
So next time we have again QueryBuilder, which builds DQL, then it's returned by cache.

**Comment by greg0ire** [comment]
Hmmm yes I think you're right, it's probably executed every time indeed 👍
Anyway, not significant at all IMO.

**Review by SenseException — APPROVED** [review]
The pattern is not a expensive one. I think this should be fine.

**Comment by greg0ire** [comment]
Thanks @Warxcell ! I squashed your commits because the first one and the last one really should, and although you made an argument for squashing in another PR, in this case it's unlikely to apply.

**Pull request merged**